multitenancy/src/main/java/org/axonframework/extensions/multitenancy/components/deadletterqueue/MultiTenantDeadLetterQueue.java

```diff
@@ -0,0 +1,406 @@
+        R res = new TenantWrappedTransactionManager(NoTransactionManager.INSTANCE,
+                                                  tenantDescriptor).fetchInTransaction(() -> function.apply(
+                getTenantSegment(tenantDescriptor)));
+        return res;
+    }
+
+    private void executeForTenantSegment(TenantDescriptor tenantDescriptor,
+                                         Consumer<SequencedDeadLetterQueue<M>> consumer) {
+        new TenantWrappedTransactionManager(NoTransactionManager.INSTANCE, tenantDescriptor).fetchInTransaction(() -> {
+            consumer.accept(getTenantSegment(tenantDescriptor));
+            return null;
+        });
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public Registration registerTenant(TenantDescriptor tenantDescriptor) {
+        tenants.add(tenantDescriptor);
+        return () -> {
+            tenants.remove(tenantDescriptor);
+            tenantSegments.remove(tenantDescriptor);
+            return true;
+        };
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public Registration registerAndStartTenant(TenantDescriptor tenantDescriptor) {
+        return registerTenant(tenantDescriptor);
+    }
+
+    /**
+     * Return processing group that this queue is bounded to.
+     *
+     * @return processing group that this queue is bounded to.
+     */
+    public String processingGroup() {
+        return processingGroup;
+    }
+
+    /**
+     * Builder class to instantiate a {@link MultiTenantDeadLetterQueue}.
+     *
+     * @param <M> the type of {@link EventMessage} contained in the {@link DeadLetter}
+     */
+    public static class Builder<M extends EventMessage<?>> {
+
+        public TargetTenantResolver<M> targetTenantResolver;
+        public String processingGroup;
+
+        /**
+         * Sets the {@link TargetTenantResolver} used to resolve correct tenant segment based on {@link Message}
+         * message
+         *
+         * @param targetTenantResolver used to resolve correct tenant segment based on {@link Message} message
+         * @return the current Builder instance, for fluent interfacing
+         */
+        public MultiTenantDeadLetterQueue.Builder<M> targetTenantResolver(
+                TargetTenantResolver<M> targetTenantResolver) {
+            BuilderUtils.assertNonNull(targetTenantResolver, "The TargetTenantResolver is a hard requirement");
+            this.targetTenantResolver = targetTenantResolver;
+            return this;
+        }
+
+        /**
+         * Sets the processing group that this queue is bounded to.
+         *
+         * @param processingGroup the processing group that this queue is bounded to.
+         * @return the current Builder instance, for fluent interfacing
+         */
+        public MultiTenantDeadLetterQueue.Builder<M> processingGroup(String processingGroup) {
+            BuilderUtils.assertNonNull(processingGroup, "The processingGroup is a hard requirement");
```

Comment:
I would use `BuilderUtils#assertNonEmpty` method here, as we don't want users to provide `""` either.